src/pages/workspace/expensifyCard/WorkspaceEditCardLimitPage.tsx

```diff
@@ -0,0 +1,153 @@
+    const updateCardLimit = (newLimit: string) => {
+        setIsConfirmModalVisible(false);
+
+        // TODO: add API call when it's supported https://github.com/Expensify/Expensify/issues/407831
+
+        Navigation.goBack();
+    };
+
+    const submit = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM>) => {
+        const currentLimit = card.nameValuePairs?.limit ?? 0;
+        const currentSpend = currentLimit - card.availableSpend;
+        const newLimit = Number(values[INPUT_IDS.LIMIT]) * 100;
+        const newAvailableSpend = newLimit - currentSpend;
+
+        if (newAvailableSpend <= 0) {
+            setIsConfirmModalVisible(true);
+            return;
+        }
+
+        updateCardLimit(values[INPUT_IDS.LIMIT]);
+    };
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM> => {
+            const errors = ValidationUtils.getFieldRequiredErrors(values, [INPUT_IDS.LIMIT]);
+
+            // We only want integers to be sent as the limit
+            if (!Number(values.limit) || !Number.isInteger(Number(values.limit))) {
+                errors.limit = translate('iou.error.invalidAmount');
+            }
+
+            return errors;
+        },
+        [translate],
+    );
+
+    return (
+        <AccessOrNotFoundWrapper
+            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
+            policyID={policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_EXPENSIFY_CARDS_ENABLED}
+        >
+            <ScreenWrapper
+                testID={WorkspaceEditCardLimitPage.displayName}
+                shouldEnablePickerAvoiding={false}
+                shouldEnableMaxHeight
+            >
+                <HeaderWithBackButton title={translate('workspace.expensifyCard.cardLimit')} />
+                <FormProvider
+                    formID={ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM}
+                    submitButtonText={translate('common.save')}
+                    onSubmit={submit}
+                    style={styles.flex1}
+                    submitButtonStyles={[styles.mh5, styles.mt0]}
+                    submitFlexEnabled={false}
+                    enabledWhenOffline
+                    validate={validate}
+                >
+                    {({inputValues}) => (
+                        <>
+                            <InputWrapper
+                                InputComponent={AmountForm}
+                                defaultValue={CurrencyUtils.convertToFrontendAmountAsString(card.nameValuePairs?.limit, CONST.CURRENCY.USD, false)}
+                                isCurrencyPressable={false}
+                                inputID={INPUT_IDS.LIMIT}
+                                ref={inputCallbackRef}
+                            />
+                            <ConfirmModal
```

Comment:
NAB, It is better to move `ConfirmModal` outside the form. But the current code also works well